src/components/forms/FormFactory.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormGroup, Col, ControlLabel, HelpBlock, Form, FieldLevelHelp } from 'patternfly-react';
+import { get, has } from 'lodash';
+import { TextArea, Dropdown, Checkbox, Text } from '.';
+
+export const FormFactory = ({ fields, fieldsValues, onFormChange }) => {
+  const formGroups = Object.keys(fields)
+    .filter(key => !(fields[key].isVisible && fields[key].isVisible(fieldsValues) === false))
+    .map(key => {
+      let child;
+      const validMsg = get(fieldsValues[key], 'validMsg');
+      switch (fields[key].type) {
+        case 'textarea':
+          child = <TextArea fieldKey={key} value={get(fieldsValues[key], 'value', '')} onChange={onFormChange} />;
+          break;
+        case 'dropdown':
+          child = (
+            <Dropdown
+              fieldKey={key}
+              value={has(fieldsValues[key], 'value') ? fieldsValues[key].value : fields[key].default}
+              choices={typeof fields[key].values === 'function' ? fields[key].values() : fields[key].values}
+              onChange={onFormChange}
+            />
+          );
+          break;
+        case 'checkbox':
+          child = (
+            <Checkbox
+              fieldKey={key}
+              checked={get(fieldsValues[key], 'value', false)}
+              title={fields[key].title}
+              onChange={onFormChange}
+            />
+          );
+          break;
+        default:
```

Comment:
nitpick to be genric: introduce specific `type` to be same as this `default`, like:
```
case 'text':
default:
```

Reply:
fixed